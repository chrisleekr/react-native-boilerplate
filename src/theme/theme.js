import {Dimensions} from 'react-native';
import {colors} from './colors';
const {width, height} = Dimensions.get('window');

// Themes:- Can alter values here. Can only be consumed through Context (see useTheme.js file)
const themes = {
  light: {
    name: 'light',
    barStyle: 'light-content',
    color: '#000000',
    labelColor: '#525E75',
    invertColor: '#ffffff',
    layoutBg: '#f0f0f0',
    headerBg: '#f6f7f8',
    cardBg: '#e4dfa7',
    accent: '#0071ff',
    error: '#B00020',
    errorText: '#ed4337',
    lightBg: '#F5EE9E',
    borderColor: '#B00020',
    buttonActiveColor: '#6200EE',
    buttonDisabledColor: 'rgba(0, 0, 0, 0.26)',
    card: {
      shadowColor: '#000000',
      borderColor: '#B00020'
    },
    ...colors
  },
  dark: {
    name: 'dark',
    barStyle: 'dark-content',
    color: '#ffffff',
    labelColor: '#F1DDBF',
    invertColor: '#000000',
    layoutBg: '#121212',
    headerBg: '#1e1e1e',
    cardBg: '#3f4045',
    accent: '#0071ff',
    error: '#B00020',
    errorText: '#ed4337',
    lightBg: '#F5EE9E',
    borderColor: '#ffffff',
    buttonActiveColor: '#BB86FC',
    buttonDisabledColor: 'rgba(255,255,255, 0.26)',
    card: {
      shadowColor: '#121212',
      borderColor: '#ffffff'
    },
    ...colors
  }
};

const window = {
  // app dimensions
  width,
  height
};

export {themes, window};
