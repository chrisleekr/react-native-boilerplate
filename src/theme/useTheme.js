/**
 * For Dark/Light Mode
 * @exports ThemeContext, ThemeProvider, useTheme
 */

import React, {useState, createContext, useContext} from 'react';
import {Appearance} from 'react-native';

// Import preconfigured themes from theme file
import {themes} from './theme';

// Context
export const ThemeContext = createContext(null);

// Provider to be used in index/App/or top of any parent
export const ThemeProvider = ({children}) => {
  const colorScheme = Appearance.getColorScheme();

  const [theme, setTheme] = useState(themes[colorScheme]);
  return (
    <ThemeContext.Provider value={{theme, setTheme}}>
      {children}
    </ThemeContext.Provider>
  );
};

// useTheme hook for easy access to theme and setTheme
export const useTheme = () => {
  const {theme, setTheme} = useContext(ThemeContext);

  const toggleTheme = v => {
    setTheme(v ? themes.dark : themes.light);
  };

  return {theme, toggleTheme};
};
