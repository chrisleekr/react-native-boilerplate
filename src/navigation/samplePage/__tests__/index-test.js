import React from 'react';
import {NavigationContainer} from '@react-navigation/native';

import {cleanup, render} from '@testing-library/react-native';

import {ThemeProvider} from '../../../theme/useTheme';

import SamplePageNavigation from '../index';

let rendered;

jest.mock('../../../component/button/headerLeftButton', () => props => (
  <header-left testID="header-left" {...props} />
));
jest.mock('../../../screen/samplePage', () => props => (
  <sample-page-screen testID="sample-page-screen" {...props} />
));

describe('SamplePageNavigation', () => {
  afterEach(cleanup);

  beforeEach(() => {
    rendered = render(
      <ThemeProvider>
        <NavigationContainer>
          <SamplePageNavigation />
        </NavigationContainer>
      </ThemeProvider>
    );
  });

  it('matches snapshot', () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('renders header left', () => {
    expect(rendered.queryByTestId('header-left')).toBeTruthy();
  });

  it('renders sample page screen', () => {
    expect(rendered.queryByTestId('sample-page-screen')).toBeTruthy();
  });
});
