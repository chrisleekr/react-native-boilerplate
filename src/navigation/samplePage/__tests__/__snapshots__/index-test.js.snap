// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SamplePageNavigation matches snapshot 1`] = `
<RNSScreenStack
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNSScreen
    collapsable={false}
    forwardedRef={[Function]}
    onAppear={[Function]}
    onDisappear={[Function]}
    onDismissed={[Function]}
    onTransitionProgress={[Function]}
    onWillDisappear={[Function]}
    replaceAnimation="push"
    stackAnimation="slide_from_right"
    stackPresentation="push"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RNSScreenStackHeaderConfig
      backButtonInCustomView={false}
      backgroundColor="#f6f7f8"
      color="#000000"
      direction="ltr"
      disableBackButtonMenu={false}
      hidden={false}
      hideBackButton={false}
      largeTitleHideShadow={false}
      title="Sample Pages"
      titleColor="#000000"
      titleFontWeight="normal"
      topInsetEnabled={false}
      translucent={false}
    >
      <RNSScreenStackHeaderSubview
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        type="left"
      >
        <header-left
          navigation={
            Object {
              "addListener": [Function],
              "canGoBack": [Function],
              "dispatch": [Function],
              "getId": [Function],
              "getParent": [Function],
              "getState": [Function],
              "goBack": [Function],
              "isFocused": [Function],
              "navigate": [Function],
              "pop": [Function],
              "popToTop": [Function],
              "push": [Function],
              "removeListener": [Function],
              "replace": [Function],
              "reset": [Function],
              "setOptions": [Function],
              "setParams": [Function],
            }
          }
          testID="header-left"
        />
      </RNSScreenStackHeaderSubview>
    </RNSScreenStackHeaderConfig>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
          },
          undefined,
        ]
      }
    >
      <sample-page-screen
        navigation={
          Object {
            "addListener": [Function],
            "canGoBack": [Function],
            "dispatch": [Function],
            "getId": [Function],
            "getParent": [Function],
            "getState": [Function],
            "goBack": [Function],
            "isFocused": [Function],
            "navigate": [Function],
            "pop": [Function],
            "popToTop": [Function],
            "push": [Function],
            "removeListener": [Function],
            "replace": [Function],
            "reset": [Function],
            "setOptions": [Function],
            "setParams": [Function],
          }
        }
        route={
          Object {
            "key": "SamplePageScreen-routeUniqId",
            "name": "SamplePageScreen",
            "params": undefined,
          }
        }
        testID="sample-page-screen"
      />
    </View>
  </RNSScreen>
</RNSScreenStack>
`;
